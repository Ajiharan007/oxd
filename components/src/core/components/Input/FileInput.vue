<template>
  <input
    type="file"
    ref="input"
    :class="fileInputClasses"
    @focus="onFocus"
    @blur="onBlur"
    @input="onInput"
  />
  <div :class="classes" :style="style" @click="onClick">{{ inputValue }}</div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export interface State {
  focused: boolean;
  inputValue: string;
}

export default defineComponent({
  name: 'oxd-file-input',

  props: {
    style: {
      type: Object,
    },
    hasError: {
      type: Boolean,
      default: false,
    },
  },

  data(): State {
    return {
      focused: false,
      inputValue: '',
    };
  },

  computed: {
    classes(): object {
      return {
        'oxd-input': true,
        'oxd-file-div': true,
        'oxd-input--active': !this.focused,
        'oxd-input--focus': this.focused,
        'oxd-input--error': this.hasError,
      };
    },
    fileInputClasses(): object {
      return {
        'oxd-file-input': true,
      };
    },
  },

  methods: {
    onClick(e: Event) {
      const inputRef = this.$refs.input as HTMLInputElement;
      inputRef.focus();
      inputRef.click();
      this.$emit('click', e);
    },
    onFocus(e: Event) {
      this.focused = true;
      this.$emit('focus', e);
    },
    onBlur(e: Event) {
      this.focused = false;
      this.$emit('blur', e);
    },
    onInput(e: Event) {
      const files = [...((e.target as HTMLInputElement).files || [])];
      this.inputValue = files.map((file: File) => file.name).join(', ');
      this.$emit('input', e);
    },
  },
});
</script>

<style src="./input.scss" lang="scss" scoped></style>
