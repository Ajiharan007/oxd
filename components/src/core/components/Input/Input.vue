<template>
  <input
    :class="classes"
    :style="style"
    :value="label"
    @focus="onFocus"
    @blur="onBlur"
  />
</template>

<script lang="ts">
export default {
  name: 'oxd-input',

  props: {
    label: {
      type: String,
      required: true,
    },
    style: {
      type: Object,
    },
  },

  data() {
    return {
      focused: false as boolean,
    };
  },

  computed: {
    classes() {
      return {
        'oxd-input': true,
        'oxd-input--active': !this.focused,
        'oxd-input--focus': this.focused,
      };
    },
  },

  methods: {
    onFocus(e: Event) {
      this.focused = true;
      this.$emit('focus', e);
    },
    onBlur(e: Event) {
      this.focused = false;
      this.$emit('blur', e);
    },
  },
};
</script>

<style src="./input.scss" lang="scss" scoped></style>
