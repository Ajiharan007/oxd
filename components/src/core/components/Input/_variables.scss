$oxd-input-min-height: 45px !default;
