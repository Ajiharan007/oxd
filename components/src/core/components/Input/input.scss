@import "../../../styles";
@import "variables";

.oxd-input {
  @include oxd-input-control;
  min-height: $oxd-input-min-height;
}

.oxd-input--active {
  border: 1px solid #e8eaef;
  box-shadow: 1px 1px 6px rgba(35, 35, 36, 0.12);
}

.oxd-input--focus {
  border: 1px solid #929baa;
  box-shadow: 1px 1px 6px rgba(35, 35, 36, 0.12);
}
