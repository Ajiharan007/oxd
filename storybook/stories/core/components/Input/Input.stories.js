import Input from '@orangehrm/oxd/core/components/Input/Input';

export default {
  title: 'Example/Input',
  component: Input,
  argTypes: {
    style: {control: {type: 'object'}},
  },
};

const Template = (args, {argTypes}) => ({
  props: Object.keys(argTypes),
  components: {'oxd-input': Input},
  template: '<oxd-input v-bind="$props" />',
});

export const Default = Template.bind({});
Default.args = {
  label: 'Input',
};

export const Colored = Template.bind({});
Colored.args = {
  label: 'Input',
  style: {backgroundColor: 'aliceblue'},
};
