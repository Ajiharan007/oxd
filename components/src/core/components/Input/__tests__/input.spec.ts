import {mount} from '@vue/test-utils';
import Input from '@orangehrm/oxd/core/components/Input/Input.vue';

describe('Input.vue', () => {
  it('renders OXD Input', () => {
    const label = 'Input';
    const wrapper = mount(Input, {
      props: {label},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
