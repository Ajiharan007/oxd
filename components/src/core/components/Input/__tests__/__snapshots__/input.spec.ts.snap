// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue renders OXD Input 1`] = `<input class="oxd-input oxd-input--active">`;
